File: Sorting-Algs/Javascript/Quick-Sort.js
const fs = require('fs');

let array = fs.readFileSync('../data.json', 'utf8');

array = JSON.parse(array)


function quickSort(arr) {
    if (arr.length <= 1) {
        return arr;
    }

    const pivot = arr[arr.length - 1];
    const left = [];
    const right = [];

    for (let i = 0; i < arr.length - 1; i++) {
        if (arr[i] <= pivot) {
            left.push(arr[i]);
        } else {
            right.push(arr[i]);
        }
    }

    return [...quickSort(left), pivot, ...quickSort(right)];
}

const start = process.hrtime.bigint();
quickSort(array);
const end = process.hrtime.bigint();
const duration = end - start;
console.log(`Execution time: ${duration} ns`);

File: Sorting-Algs/Python/Quick-Sort.py
import time
import json

def quick_sort(arr):
    if len(arr) <= 1:
        return

    pivot = arr[-1]
    i = 0
    for j in range(len(arr) - 1):
        if arr[j] <= pivot:
            arr[i], arr[j] = arr[j], arr[i]
            i += 1

    arr[i], arr[-1] = arr[-1], arr[i]

    quick_sort(arr[:i])
    quick_sort(arr[i + 1:])

f = open("../data.json", "r")
array = json.loads(f.read())


start = time.time_ns()
quick_sort(array)
duration = time.time_ns() - start
print(f"Execution time: {duration} ns")